### PR TITLE
feat: implement FetchPolicy for prompt retrieval

### LINKED ISSUE #966: Implement FetchPolicy for TypeScript SDK
## Parent Issue
Closes part of #840

## Summary
Add `FetchPolicy` enum and support to `prompts.get()` in the TypeScript SDK.

## Fetch Policies to Implement

| Policy | Behavior | Use Case |
|--------|----------|----------|
| `MATERIALIZED_FIRST` | Local → API fallback | **Default** (current behavior) |
| `ALWAYS_FETCH` | API → Local fallback | Live updates + offline fallback |
| `CACHE_TTL` | API every X min, local between | Controlled update frequency |
| `MATERIALIZED_ONLY` | Local only, throw if missing | Air-gapped environments |

## API Design

```typescript
import { FetchPolicy } from "langwatch";

// Default behavior (MATERIALIZED_FIRST)
const prompt = await langwatch.prompts.get("my-prompt");

// Explicit policy
const prompt = await langwatch.prompts.get("my-prompt", {
  fetchPolicy: FetchPolicy.ALWAYS_FETCH
});

// With TTL
const prompt = await langwatch.prompts.get("my-prompt", {
  fetchPolicy: FetchPolicy.CACHE_TTL,
  cacheTtlMinutes: 5
});
```

## Tasks
- [ ] Add `FetchPolicy` enum to prompts types
- [ ] Update `PromptsFacade.get()` signature to accept `fetchPolicy` option
- [ ] Implement `MATERIALIZED_FIRST` logic (already exists, make explicit)
- [ ] Implement `ALWAYS_FETCH` logic
- [ ] Implement `MATERIALIZED_ONLY` logic
- [ ] Implement `CACHE_TTL` logic with in-memory cache
- [ ] Export `FetchPolicy` from top-level `langwatch` module
- [ ] Add unit tests for each policy
- [ ] Update docs to uncomment fetch policy section

## Acceptance Criteria
- All 4 fetch policies work as documented
- Default behavior unchanged (MATERIALIZED_FIRST)
- Tests cover all policies and edge cases

### PR DESCRIPTION
**Pull request by drewdrewthis**
- Add FetchPolicy enum with MATERIALIZED_FIRST, ALWAYS_FETCH, CACHE_TTL, MATERIALIZED_ONLY strategies
- Implement fetch policy logic in PromptsFacade with caching and fallback behavior
- Add integration tests and E2E test coverage for all fetch policies
- Export FetchPolicy and GetPromptOptions types
- Add BDD spec file for prompt retrieval scenarios
- Add Cursor rules for BDD workflow guidance

# Related Issue

- Resolve #966

### CODE REVIEW

File: typescript-sdk/src/client-sdk/services/prompts/prompts.facade.ts [hunk 1]
```diff
@@ -52,6 +93,55 @@ export class PromptsFacade {
...
+      this.cache.set(handleOrId, { data: serverPrompt, timestamp: now });
+      return new Prompt(serverPrompt);
+    } catch {
+      const localPrompt = await this.localPromptsService.get(handleOrId);
+      if (localPrompt) {
+        return new Prompt(localPrompt);
+      }
+      throw new Error(`Prompt "${handleOrId}" not found locally or on server`);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (drewdrewthis) [reply typescript-sdk/src/client-sdk/services/prompts/prompts.facade.ts:151]: Valid concerns

1. It's okay if we have different caches for id and handle, that would be otherwise confusing to manage for the user
2. Unnecessary optimization. It's more of a policy than a real dedicated caching system - the behavior is upheld
3. same

File: typescript-sdk/__tests__/e2e/prompts/fetch-policy.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        return runMaterializedOnlyPolicy(handle, cli);
+      });
+      expect(prompt).toBeTruthy();
+      expect(prompt.handle).toBe(handle);
+      expect(prompt.prompt ?? "").toContain(content);
+      expect(calls).toBe(0);
+    } finally {
+      temp.dispose();
+    }
+  }, 60_000);
```
[the remark this answers is not in the document]
Comment (drewdrewthis) [reply typescript-sdk/__tests__/e2e/prompts/fetch-policy.e2e.test.ts:106]: Ephemeral DB -- not worth the extra time in the CI

File: typescript-sdk/src/client-sdk/services/prompts/prompts.facade.ts [hunk 3]
```diff
@@ -52,6 +93,63 @@ export class PromptsFacade {
...
+
+  private async getMaterializedOnly(handleOrId: string): Promise<Prompt> {
+    const localPrompt = await this.localPromptsService.get(handleOrId);
+    if (localPrompt) {
+      return new Prompt(localPrompt);
+    }
+    throw new Error(`Prompt "${handleOrId}" not found in materialized files`);
+  }
```
[the remark this answers is not in the document]
Comment (drewdrewthis) [reply typescript-sdk/src/client-sdk/services/prompts/prompts.facade.ts:118]: low priority

File: typescript-sdk/AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Agent Guidelines for TypeScript SDK
+
+## Common Mistakes | Correct Behavior
+
+| Common Mistake | Correct Behavior |
+|----------------|------------------|
+| Putting method-specific options interfaces in `types.ts` | Options interfaces belong in the file that uses them (e.g., `GetPromptOptions` in `prompts.facade.ts`) |
+| Using "should" in test descriptions | Use action-based descriptions: `it("checks local first")` not `it("should check local first")` |
+| Creating shared types for single-use interfaces | Colocate interfaces with their usage; only extract to `types.ts` when shared across multiple files |
```
Comment (rogeriochaves) [inline typescript-sdk/AGENTS.md:9]: very good too!

File: typescript-sdk/examples/prompt-management/fetch-policy.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  cacheTtlMinutes = 5,
+) => {
+  const { LangWatch, FetchPolicy } = await import("../../dist");
+
+  const langwatch = new LangWatch();
+  return langwatch.prompts.get(handle, {
+    fetchPolicy: FetchPolicy.CACHE_TTL,
+    cacheTtlMinutes,
+  });
+};
```
Comment (rogeriochaves) [inline typescript-sdk/examples/prompt-management/fetch-policy.ts:67]: hmmm why is this file here on examples? It's a bit weird
examples are supposed to resemble real-world usage of the features, apps you can actually run, but this one doesn't seem much different than the tests actually, I don't think there is even a way to execute this file right?

I'd either move into having a somewhate more resemble real world usage of prompts in an aplication, or just remove this example and helpers for now
